### ISSUE TITLE
we should convert the solver to using pressure as primary variable in the fluid parts of the model

### ISSUE THREAD

**Issue by komatits** [geodynamics/specfem3d#748]
- Daniel @danielpeter you are of course right that the formulation is not self adjoint in the fluid, contrary to the solid: eqs A22 and A23 of Luo et al (2013) show that clearly; I had not realized that when I started to replace the variable names with a script, thanks for letting me know; I have thus canceled these Git pull requests

- however, with Vadim @vmont yesterday we checked and showed that it would be easy to convert the solver to using pressure, i.e. use pressure = Chi (instead of the current pressure = - Chi_dot_dot); coupling with a solid in a non-iterative way would still work fine, we checked that (I previously and erroneously thought it was not possible)

- this would have the HUGE advantage that pressure is the natural variable in ocean acoustics, in medical imaging, also in the oil industry

- another advantage would be that sources would be given in pressure rather than in a weird potential

- yet another advantage is that USE_TRICK_FOR_BETTER_PRESSURE would not be needed any more, it would be done automatically and pressure would automatically be accurate

- yet yet another advantage is that in the last two lines of eq A9 of Luo et al (2013) the two lines would be in pressure, rather than a mixture of pressure and Chi as in the current form of A9, which is weird

- however, as you say if we do so we need to change the adjoint part accordingly, based on a modified version of eqs A22 and A23 in Luo et al (2013); this is not too difficult to do, but not done yet (Vadim would need to write the new eqs for A22 and A23 modified accordingly)

- textbook velocity or displacement potentials do not exist in fluids with spatially-varying density, but potentials of rho times that (as we currently use) do exist, since they correspond to pressure; thus he agrees with us, and we are all set; he just thinks that the word 'potential' could be/is misleading for people, and 'pressure' is much clearer.

To summarize:

1/ we should switch to pressure in the forward solver for sure

2/ this implies modifying the adjoint as well, as mentioned by Daniel, i.e. writing a new form of eqs A22 and A23 (and also A9) and programming that in the adjoint part of the code

3/ thus, we will do it, but only in the second part of 2016, probably this fall or even this winter (Vadim is busy with other things for now)    

CC'ing @bottero @pcristini @vmont @danielpeter @xiezhinan @EtienneBachmann

**Comment by komatits** [comment]
After discussion with Vadim @vmont and Etienne @EtienneBachmann we realized that is probably not needed, thus closing the issue.

**Comment by komatits** [comment]
From  @EtienneBachmann, to me and Jeroen @jeroentromp and @vmont and @danielpeter :

I suppose the most important for the choice between Chi and pressure model is not the adjoint source, but the forward one, as long as we primarily need to describe as accurately as possible the behavior of transducers/geophones, which leads to the Chi model.
